src/scheduler/workers/unpublish/delete/index.ts

```diff
@@ -0,0 +1,90 @@
+import { startOfToday, subYears } from "date-fns";
+import { prisma } from "scheduler/prismaClient";
+import { schedulerLogger } from "scheduler/logger";
+
+export default async function deleteItemsAfterAYear() {
+  const logger = schedulerLogger.child({ method: "deleteItemsAfterAYear" });
+
+  try {
+    const dateOneYearAgo = subYears(startOfToday(), 1);
+    const itemsUnpublishedByAR = await prisma.listItem.findMany({
+      where: {
+        status: "ANNUAL_REVIEW_OVERDUE",
+        isAnnualReview: false,
+        isPublished: false,
+        AND: [
+          {
+            history: {
+              some: {
+                type: "ANNUAL_REVIEW_OVERDUE",
+                time: {
+                  lte: dateOneYearAgo,
+                },
+              },
+            },
+          },
+          {
+            history: {
+              some: {
+                type: "UNPUBLISHED",
+                time: {
+                  lte: dateOneYearAgo,
+                },
+              },
+            },
+          },
+        ],
+      },
+      include: {
+        history: {
+          orderBy: {
+            time: "desc",
+          },
+        },
+      },
+    });
+
+    const listItemsToDelete = itemsUnpublishedByAR.filter((item) => {
+      const unpublishedIndex = item.history.findIndex((history) => history.type === "UNPUBLISHED");
```

Comment:
the filter shouldn't be necessary

this can still be done by query. we just want to check that something hasn't been republished since it was unpublished. (see my prev comment) 

```
{
        history: { // double check a user hasn't been re-published between a year ago and today.
          none: {
            type: "PUBLISHED",
            time: {
              gte: dateOneYearAgo, 
            },
          },
        },
      },
```

the full query would be checking 
- that there has been an "ANNUAL_REVIEW_OVERDUE" event that happened 28-06-2022 or before (lte) AND
- that there are NONE "PUBLISHED" events that happened 28-06-2022 or after (gte) 

both must qualify for a row to be returned (so no need to filter).